src/approaches/__init__.py

```diff
@@ -1,67 +1,33 @@
-"""Default imports for approaches folder."""
+"""Handle creation of approaches."""
 
-from typing import Set, List
+import pkgutil
+from typing import Set, List, TYPE_CHECKING
 from gym.spaces import Box
 from predicators.src.approaches.base_approach import BaseApproach, \
     ApproachTimeout, ApproachFailure
-from predicators.src.approaches.random_actions_approach import \
-    RandomActionsApproach
-from predicators.src.approaches.random_options_approach import \
-    RandomOptionsApproach
-from predicators.src.approaches.gnn_policy_approach import \
-    GNNPolicyApproach
-from predicators.src.approaches.bilevel_planning_approach import \
-    BilevelPlanningApproach
-from predicators.src.approaches.oracle_approach import OracleApproach
-from predicators.src.approaches.nsrt_learning_approach import \
-    NSRTLearningApproach
-from predicators.src.approaches.interactive_learning_approach import \
-    InteractiveLearningApproach
-from predicators.src.approaches.grammar_search_invention_approach import \
-    GrammarSearchInventionApproach
 from predicators.src.structs import Predicate, ParameterizedOption, \
     Type, Task
+from predicators.src import utils
 
-__all__ = [
-    "BaseApproach",
-    "OracleApproach",
-    "RandomActionsApproach",
-    "RandomOptionsApproach",
-    "GNNPolicyApproach",
-    "BilevelPlanningApproach",
-    "NSRTLearningApproach",
-    "InteractiveLearningApproach",
-    "GrammarSearchInventionApproach",
-    "ApproachTimeout",
-    "ApproachFailure",
-]
+__all__ = ["BaseApproach", "ApproachTimeout", "ApproachFailure"]
+
+if not TYPE_CHECKING:
```

Comment:
What happens if we do it if we're type checking? is it just slower?

Reply:
it crashes and burns horrendously LOL, mypy starts to search over all the modules that are imported in the whole codebase for some reason...